### PR TITLE
Fixed docstring errors inside torch/cuda/ and torch/optim/ (Docathon H2)

### LINKED ISSUE #112592: Fix docstring errors in adagrad.py, asgd.py, sparse_adam.py, adam.py, adamax.py, grad_scaler.py, adamw.py, random.py, _functional.py, lbfgs.py, adadelta.py, __init__.py, autocast_mode.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/cuda/random.py`, **Entity**: `get_rng_state`, **Line**: 13, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/random.py`, **Entity**: `get_rng_state_all`, **Line**: 35, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/cuda/random.py`, **Entity**: `get_rng_state_all`, **Line**: 35, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/random.py`, **Entity**: `set_rng_state`, **Line**: 44, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `set_rng_state_all`, **Line**: 69, **Description**: Docstring is over-indented
- **File**: `torch/cuda/random.py`, **Entity**: `set_rng_state_all`, **Line**: 69, **Description**: Multi-line docstring closing quotes should be on a separate line
- **File**: `torch/cuda/random.py`, **Entity**: `set_rng_state_all`, **Line**: 69, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `set_rng_state_all`, **Line**: 69, **Description**: Section has no content ('Args')
- **File**: `torch/cuda/random.py`, **Entity**: `manual_seed`, **Line**: 78, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/random.py`, **Entity**: `manual_seed`, **Line**: 78, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `manual_seed_all`, **Line**: 100, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/random.py`, **Entity**: `manual_seed_all`, **Line**: 100, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `seed`, **Line**: 118, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/random.py`, **Entity**: `seed`, **Line**: 118, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `seed_all`, **Line**: 135, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/random.py`, **Entity**: `seed_all`, **Line**: 135, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/cuda/random.py`, **Entity**: `initial_seed`, **Line**: 155, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `autocast`, **Line**: 14, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_fwd`, **Line**: 75, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_fwd`, **Line**: 75, **Description**: First line should end with a period (not 'f')
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_fwd`, **Line**: 75, **Description**: First line should be in imperative mood; try rephrasing (found 'Helper')
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_bwd`, **Line**: 114, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_bwd`, **Line**: 114, **Description**: First line should end with a period (not 'f')
- **File**: `torch/cuda/amp/autocast_mode.py`, **Entity**: `custom_bwd`, **Line**: 114, **Description**: First line should be in imperative mood; try rephrasing (found 'Helper')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `_MultiDeviceReplicator`, **Line**: 14, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `_MultiDeviceReplicator`, **Line**: 14, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `step`, **Line**: 294, **Description**: First line should end with a period (not '
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `update`, **Line**: 381, **Description**: First line should be in imperative mood (perhaps 'Update', not 'Updates')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_scale`, **Line**: 449, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_growth_factor`, **Line**: 461, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_growth_factor`, **Line**: 461, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_growth_factor`, **Line**: 467, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_growth_factor`, **Line**: 467, **Description**: First line should end with a period (not '
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_backoff_factor`, **Line**: 474, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_backoff_factor`, **Line**: 474, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_backoff_factor`, **Line**: 480, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_backoff_factor`, **Line**: 480, **Description**: First line should end with a period (not '
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_growth_interval`, **Line**: 487, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `get_growth_interval`, **Line**: 487, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_growth_interval`, **Line**: 493, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `set_growth_interval`, **Line**: 493, **Description**: First line should end with a period (not '
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `is_enabled`, **Line**: 506, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `is_enabled`, **Line**: 506, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `state_dict`, **Line**: 512, **Description**: First line should end with a period (not '
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `state_dict`, **Line**: 512, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/cuda/amp/grad_scaler.py`, **Entity**: `load_state_dict`, **Line**: 534, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/_functional.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lbfgs.py`, **Entity**: `LBFGS`, **Line**: 185, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lbfgs.py`, **Entity**: `LBFGS`, **Line**: 185, **Description**: First line should end with a period (not 'c')
- **File**: `torch/optim/lbfgs.py`, **Entity**: `step`, **Line**: 285, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/sparse_adam.py`, **Entity**: `step`, **Line**: 40, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adadelta.py`, **Entity**: `step`, **Line**: 79, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adadelta.py`, **Entity**: `adadelta`, **Line**: 188, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/adagrad.py`, **Entity**: `step`, **Line**: 102, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adagrad.py`, **Entity**: `adagrad`, **Line**: 201, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/adam.py`, **Entity**: `step`, **Line**: 124, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adam.py`, **Entity**: `adam`, **Line**: 267, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/adam.py`, **Entity**: `adam`, **Line**: 267, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/adamax.py`, **Entity**: `step`, **Line**: 87, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adamax.py`, **Entity**: `adamax`, **Line**: 197, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/adamw.py`, **Entity**: `step`, **Line**: 151, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/adamw.py`, **Entity**: `adamw`, **Line**: 302, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/asgd.py`, **Entity**: `step`, **Line**: 98, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/asgd.py`, **Entity**: `asgd`, **Line**: 181, **Description**: No blank lines allowed after function docstring (found 1)

cc @carljparker

### PR DESCRIPTION
**Pull request by pilot-j**
Fixes #112592 
1) **File: torch/cuda/random.py**
```
Before:
/content/pytorch/torch/cuda/random.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/cuda/random.py:21 in public function `get_rng_state`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/random.py:43 in public function `get_rng_state_all`:
        D202: No blank lines allowed after function docstring (found 1)
/content/pytorch/torch/cuda/random.py:43 in public function `get_rng_state_all`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/random.py:54 in public function `set_rng_state`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:79 in public function `set_rng_state_all`:
        D208: Docstring is over-indented
/content/pytorch/torch/cuda/random.py:79 in public function `set_rng_state_all`:
        D209: Multi-line docstring closing quotes should be on a separate line
/content/pytorch/torch/cuda/random.py:79 in public function `set_rng_state_all`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:79 in public function `set_rng_state_all`:
        D414: Section has no content ('Args')
/content/pytorch/torch/cuda/random.py:88 in public function `manual_seed`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/random.py:88 in public function `manual_seed`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:110 in public function `manual_seed_all`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/random.py:110 in public function `manual_seed_all`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:128 in public function `seed`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/random.py:128 in public function `seed`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:146 in public function `seed_all`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/random.py:146 in public function `seed_all`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
/content/pytorch/torch/cuda/random.py:167 in public function `initial_seed`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
18
```

```
After:
/content/pytorch/torch/cuda/random.py:1 at module level:
        D100: Missing docstring in public module
1

```
2) **File: torch/cuda/amp/autocast_mode.py**
```
Before: /content/pytorch/torch/cuda/amp/autocast_mode.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/cuda/amp/autocast_mode.py:18 in public class `autocast`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/autocast_mode.py:23 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/cuda/amp/autocast_mode.py:38 in public method `__enter__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/autocast_mode.py:44 in public method `__exit__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/autocast_mode.py:49 in public method `__call__`:
        D102: Missing docstring in public method
/content/pytorch/torch/cuda/amp/autocast_mode.py:90 in public function `custom_fwd`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/autocast_mode.py:90 in public function `custom_fwd`:
        D400: First line should end with a period (not 'f')
/content/pytorch/torch/cuda/amp/autocast_mode.py:90 in public function `custom_fwd`:
        D401: First line should be in imperative mood; try rephrasing (found 'Helper')
/content/pytorch/torch/cuda/amp/autocast_mode.py:130 in public function `custom_bwd`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/autocast_mode.py:130 in public function `custom_bwd`:
        D400: First line should end with a period (not 'f')
/content/pytorch/torch/cuda/amp/autocast_mode.py:130 in public function `custom_bwd`:
        D401: First line should be in imperative mood; try rephrasing (found 'Helper')
12
```
```
After:
/content/pytorch/torch/cuda/amp/autocast_mode.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/cuda/amp/autocast_mode.py:23 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/cuda/amp/autocast_mode.py:38 in public method `__enter__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/autocast_mode.py:44 in public method `__exit__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/autocast_mode.py:49 in public method `__call__`:
        D102: Missing docstring in public method
5
```

3)  **File: torch/cuda/amp/grad_scaler.py**
```
Before: /content/pytorch/torch/cuda/amp/grad_scaler.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/cuda/amp/grad_scaler.py:17 in private class `_MultiDeviceReplicator`:
        D200: One-line docstring should fit on one line with quotes (found 3)
/content/pytorch/torch/cuda/amp/grad_scaler.py:39 in public class `OptState`:
        D101: Missing docstring in public class
/content/pytorch/torch/cuda/amp/grad_scaler.py:50 in public class `GradScaler`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/grad_scaler.py:50 in public class `GradScaler`:
        D400: First line should end with a period (not 'g')
/content/pytorch/torch/cuda/amp/grad_scaler.py:115 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/cuda/amp/grad_scaler.py:354 in public method `step`:
        D400: First line should end with a period (not ':')
/content/pytorch/torch/cuda/amp/grad_scaler.py:456 in public method `update`:
        D401: First line should be in imperative mood (perhaps 'Update', not 'Updates')
/content/pytorch/torch/cuda/amp/grad_scaler.py:529 in public method `get_scale`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:544 in public method `get_growth_factor`:
        D200: One-line docstring should fit on one line with quotes (found 3)
/content/pytorch/torch/cuda/amp/grad_scaler.py:544 in public method `get_growth_factor`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:550 in public method `set_growth_factor`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/grad_scaler.py:550 in public method `set_growth_factor`:
        D400: First line should end with a period (not ':')
/content/pytorch/torch/cuda/amp/grad_scaler.py:557 in public method `get_backoff_factor`:
        D200: One-line docstring should fit on one line with quotes (found 3)
/content/pytorch/torch/cuda/amp/grad_scaler.py:557 in public method `get_backoff_factor`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:563 in public method `set_backoff_factor`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/grad_scaler.py:563 in public method `set_backoff_factor`:
        D400: First line should end with a period (not ':')
/content/pytorch/torch/cuda/amp/grad_scaler.py:570 in public method `get_growth_interval`:
        D200: One-line docstring should fit on one line with quotes (found 3)
/content/pytorch/torch/cuda/amp/grad_scaler.py:570 in public method `get_growth_interval`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:576 in public method `set_growth_interval`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/cuda/amp/grad_scaler.py:576 in public method `set_growth_interval`:
        D400: First line should end with a period (not ':')
/content/pytorch/torch/cuda/amp/grad_scaler.py:592 in public method `is_enabled`:
        D200: One-line docstring should fit on one line with quotes (found 3)
/content/pytorch/torch/cuda/amp/grad_scaler.py:592 in public method `is_enabled`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:598 in public method `state_dict`:
        D400: First line should end with a period (not ':')
/content/pytorch/torch/cuda/amp/grad_scaler.py:598 in public method `state_dict`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
/content/pytorch/torch/cuda/amp/grad_scaler.py:624 in public method `load_state_dict`:
        D401: First line should be in imperative mood (perhaps 'Load', not 'Loads')
/content/pytorch/torch/cuda/amp/grad_scaler.py:649 in public method `__getstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/grad_scaler.py:665 in public method `__setstate__`:
        D105: Missing docstring in magic method
28
```
```
After:
/content/pytorch/torch/cuda/amp/grad_scaler.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/cuda/amp/grad_scaler.py:40 in public class `OptState`:
        D101: Missing docstring in public class
/content/pytorch/torch/cuda/amp/grad_scaler.py:117 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/cuda/amp/grad_scaler.py:647 in public method `__getstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/cuda/amp/grad_scaler.py:663 in public method `__setstate__`:
        D105: Missing docstring in magic method
5
```
4) **File: torch/optim/_functional.py**
```
Before:
/content/pytorch/torch/optim/_functional.py:1 at module level:
        D400: First line should end with a period (not 'e')
1
```
```
After:
0

```
5) **File: torch/optim/__init__.py**
```
Before:
/content/pytorch/torch/optim/__init__.py:1 at module level:
        D205: 1 blank line required between summary line and description (found 0)
1
```
```
After:
0

```
6) **File: torch/optim/lbfgs.py**
```
Before:
/content/pytorch/torch/optim/lbfgs.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/lbfgs.py:185 in public class `LBFGS`:
        D205: 1 blank line required between summary line and description (found 0)
/content/pytorch/torch/optim/lbfgs.py:185 in public class `LBFGS`:
        D400: First line should end with a period (not 'c')
/content/pytorch/torch/optim/lbfgs.py:215 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/lbfgs.py:285 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
5
```
```
After:
/content/pytorch/torch/optim/lbfgs.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/lbfgs.py:217 in public method `__init__`:
        D107: Missing docstring in __init__
2
```
7)**File: torch/optim/sparse_adam.py**
```
Before: /content/pytorch/torch/optim/sparse_adam.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/sparse_adam.py:7 in public class `SparseAdam`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/sparse_adam.py:8 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/sparse_adam.py:40 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
4
```
```
After:
/content/pytorch/torch/optim/sparse_adam.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/sparse_adam.py:7 in public class `SparseAdam`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/sparse_adam.py:8 in public method `__init__`:
        D107: Missing docstring in __init__
3
```
8) **File:torch/optim/adadelta.py**
```
Before:
/content/pytorch/torch/optim/adadelta.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adadelta.py:11 in public class `Adadelta`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adadelta.py:12 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adadelta.py:44 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adadelta.py:82 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/adadelta.py:193 in public function `adadelta`:
        D202: No blank lines allowed after function docstring (found 1)
6
```
```
After:
/content/pytorch/torch/optim/adadelta.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adadelta.py:11 in public class `Adadelta`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adadelta.py:12 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adadelta.py:44 in public method `__setstate__`:
        D105: Missing docstring in magic method
4
```
9) **File: torch/optim/adagrad.py**
```
Before:
/content/pytorch/torch/optim/adagrad.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adagrad.py:11 in public class `Adagrad`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adagrad.py:12 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adagrad.py:63 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adagrad.py:78 in public method `share_memory`:
        D102: Missing docstring in public method
/content/pytorch/torch/optim/adagrad.py:100 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/adagrad.py:201 in public function `adagrad`:
        D202: No blank lines allowed after function docstring (found 1)
7
```
```
After:
/content/pytorch/torch/optim/adagrad.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adagrad.py:11 in public class `Adagrad`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adagrad.py:12 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adagrad.py:63 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adagrad.py:78 in public method `share_memory`:
        D102: Missing docstring in public method
5
```
10) **File: torch/optim/adam.py**
```
Before:
/content/pytorch/torch/optim/adam.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adam.py:14 in public class `Adam`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adam.py:15 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adam.py:65 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adam.py:135 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/adam.py:281 in public function `adam`:
        D202: No blank lines allowed after function docstring (found 1)
/content/pytorch/torch/optim/adam.py:281 in public function `adam`:
        D205: 1 blank line required between summary line and description (found 0)
7
```
```
After:
/content/pytorch/torch/optim/adam.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adam.py:14 in public class `Adam`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adam.py:15 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adam.py:65 in public method `__setstate__`:
        D105: Missing docstring in magic method
4

```
11) **File: torch/optim/adamax.py**
```
Before:
/content/pytorch/torch/optim/adamax.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adamax.py:12 in public class `Adamax`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adamax.py:13 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adamax.py:47 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adamax.py:91 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/adamax.py:203 in public function `adamax`:
        D202: No blank lines allowed after function docstring (found 1)
6
```
```
After:
/content/pytorch/torch/optim/adamax.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adamax.py:12 in public class `Adamax`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adamax.py:13 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adamax.py:47 in public method `__setstate__`:
        D105: Missing docstring in magic method
4
```
12) **File: torch/optim/adamw.py**
```
Before:
/content/pytorch/torch/optim/adamw.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adamw.py:12 in public class `AdamW`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adamw.py:13 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adamw.py:73 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/adamw.py:153 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/adamw.py:304 in public function `adamw`:
        D202: No blank lines allowed after function docstring (found 1)
6

```
```
After:
/content/pytorch/torch/optim/adamw.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/adamw.py:12 in public class `AdamW`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/adamw.py:13 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/adamw.py:73 in public method `__setstate__`:
        D105: Missing docstring in magic method
4

```
13) **File: torch/optim/asgd.py**
```
Before:
/content/pytorch/torch/optim/asgd.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/asgd.py:17 in public class `ASGD`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/asgd.py:18 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/asgd.py:52 in public method `__setstate__`:
        D105: Missing docstring in magic method
/content/pytorch/torch/optim/asgd.py:107 in public method `step`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
/content/pytorch/torch/optim/asgd.py:195 in public function `asgd`:
        D202: No blank lines allowed after function docstring (found 1)
6
```
```
After:
/content/pytorch/torch/optim/asgd.py:1 at module level:
        D100: Missing docstring in public module
/content/pytorch/torch/optim/asgd.py:17 in public class `ASGD`:
        D101: Missing docstring in public class
/content/pytorch/torch/optim/asgd.py:18 in public method `__init__`:
        D107: Missing docstring in __init__
/content/pytorch/torch/optim/asgd.py:52 in public method `__setstate__`:
        D105: Missing docstring in magic method
4
```
Resolved docstring errors as listed. I initially changed in the main branch of forked repo which caused changes to appear in my PR to other issue. I have fixed that and hope this PR won't have any conflicts.
Kindly review @svekars @jbschlosser.
In case of any other issues please let me know. Thanks!

cc @svekars @carljparker

### CODE REVIEW

**Comment by pilot-j** [comment]
cc @svekars

File: torch/cuda/amp/grad_scaler.py [hunk 1]
```diff
@@ -541,39 +542,36 @@ def get_scale(self) -> float:
...
     def get_growth_factor(self) -> float:
-        r"""
-        Returns a Python float containing the scale growth factor.
-        """
+        r"""Return a Python float containing the scale growth factor."""
         return self._growth_factor
 
     def set_growth_factor(self, new_factor: float) -> None:
-        r"""
+        r"""Set new scale growth factor.
```
Comment (svekars) [inline torch/cuda/amp/grad_scaler.py:549]:
```suggestion
        r"""Set a new scale growth factor.
```

File: torch/cuda/amp/grad_scaler.py [hunk 2]
```diff
@@ -541,39 +542,36 @@ def get_scale(self) -> float:
...
     def get_backoff_factor(self) -> float:
-        r"""
-        Returns a Python float containing the scale backoff factor.
-        """
+        r"""Return a Python float containing the scale backoff factor."""
         return self._backoff_factor
 
     def set_backoff_factor(self, new_factor: float) -> None:
-        r"""
+        r"""Set new scale backoff factor.
```
Comment (svekars) [inline torch/cuda/amp/grad_scaler.py:561]:
```suggestion
        r"""Set a new scale backoff factor.
```

File: torch/cuda/amp/grad_scaler.py [hunk 3]
```diff
@@ -541,39 +542,36 @@ def get_scale(self) -> float:
...
     def get_growth_interval(self) -> int:
-        r"""
-        Returns a Python int containing the growth interval.
-        """
+        r"""Return a Python int containing the growth interval."""
         return self._growth_interval
 
     def set_growth_interval(self, new_interval: int) -> None:
-        r"""
+        r"""Set new growth interval.
```
Comment (svekars) [inline torch/cuda/amp/grad_scaler.py:573]:
```suggestion
        r"""Set a new growth interval.
```

**Comment by kit1980** [comment]
@pilot-j please fix the lint issues.

**Comment by svekars** [comment]
@pilot-j there is an error in the doc-build that is caused by the changed - it is unclear exactly what is causing the error. I suggest that you carefully review your changes one more time.

**Comment by pilot-j** [comment]
> @pilot-j there is an error in the doc-build that is caused by the changed - it is unclear exactly what is causing the error. I suggest that you carefully review your changes one more time.

image
It seems that the build is failing due to this link not working. On close inspection, the code written does have the correct URL(tested) with the word **S**oftware, but during build the capital **S** has been read as small case **s**. 
I am clueless on how to fix this problem. Please look into this @svekars @kit1980

**Review by kit1980 — APPROVED** [review]

**Pull request closed without merging**